Get rid of mktemp dependency to support AIX

dagwieers opened the pull request:
The function call has been renamed to better reflect what it does, and we reduced the number of calls from two to one in case the remote user is not root.

This patch also fixes a string concatenation in _copy_module() that should use os.path.join() instead.

This closes #436

This also fixes #437 for non-root remote users.

mpdehaan commented:
Thanks!

can you git rebase this?  I had to revert your mktemp patch temporarily so I am expecting to see the removal of mktemp also in this diff, but I don't see it, so I think this is just a patch on top of your original.

dagwieers commented:
I have no idea what is requested from me. I don't know Git/Github very well, I am sorry.

mpdehaan commented:
This patch does not apply to a fresh clone of devel.  The reason is I had previously reverted your previous merge commit.  I need a patch that fully implements the removal of mktemp.  Add a ".patch" to the end of the pull request URL and you will see what is and what is not in the diff.

-- Michael

On Jun 3, 2012, at 5:43 PM, Dag Wieërsreply@reply.github.com wrote:

> I have no idea what is requested from me. I don't know Git/Github very well, I am sorry.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/ansible/ansible/pull/443#issuecomment-6089360

dagwieers committed:
Get rid of mktemp dependency to support AIX

The function call has been renamed to better reflect what it does, and we
reduced the number of calls from two to one in case the remote user is not
root.

This patch also fixes a string concatenation in _copy_module() that
should use os.path.join() instead.

This closes #436

dagwieers commented:
With a little Git help from @jhoekx ...

dagwieers commented:
Is there still something wrong with this patchset ?

mpdehaan commented:
haven't gotten around to merging/testing it yet

mpdehaan commented:
merged, thanks!
